contracts/modules/cover/Cover.sol

```diff
@@ -601,19 +603,28 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon, ReentrancyGuard {
     }
 
     // increase amountPaidOut only *after* you read the segment
-    cover.amountPaidOut += SafeUintCast.toUint96(payoutAmountInAsset);
+    cover.amountPaidOut += payoutAmountInAsset.toUint96();
 
     uint allocationCount = allocations.length;
 
     for (uint i = 0; i < allocationCount; i++) {
-
       PoolAllocation memory allocation = allocations[i];
 
       uint payoutAmountInNXM = allocation.coverAmountInNXM * payoutAmountInAsset / segment.amount;
-      allocations[i].coverAmountInNXM -= SafeUintCast.toUint96(payoutAmountInNXM);
-
       uint burnAmountInNxm = payoutAmountInNXM * GLOBAL_CAPACITY_DENOMINATOR / segment.globalCapacityRatio;
-      stakingPool(i).burnStake(burnAmountInNxm);
+
+      allocations[i].coverAmountInNXM -= payoutAmountInNXM.toUint96();
```

Comment:
is this logic that was missing and it was a bug ?

Or why is it added as part of this PR

Reply:
Was showing up as a diff because the base branch wasn't nexus-v2 but a burn related PR. Changes no longer present after rebase.